src/event.rs

```diff
@@ -594,19 +594,21 @@ where
 						..
 					} => {
 						let offer_id = payment_context.offer_id;
-						let payment = PaymentDetails {
-							id: payment_id,
-							kind: PaymentKind::Bolt12Offer {
-								hash: Some(payment_hash),
-								preimage: payment_preimage,
-								secret: Some(payment_secret),
-								offer_id,
-							},
-							amount_msat: Some(amount_msat),
-							direction: PaymentDirection::Inbound,
-							status: PaymentStatus::Pending,
+						let kind = PaymentKind::Bolt12Offer {
+							hash: Some(payment_hash),
+							preimage: payment_preimage,
+							secret: Some(payment_secret),
+							offer_id,
 						};
 
+						let payment = PaymentDetails::new(
+							payment_id,
+							kind,
+							Some(amount_msat),
+							PaymentDirection::Inbound,
+							PaymentStatus::Pending,
+						);
```

Comment:
Any reason the nesting was reverted? Seems unrelated to the commit.

Reply:
Mostly difficult to review as it doesn't seem relevant to the commit. Though I wasn't sure if this was a rebase issue since a prior PR used the nested variation.